core/templates/pages/library-page/search-bar/search-bar.component.spec.ts

```diff
@@ -242,12 +242,35 @@ describe('Search bar component', () => {
     }
   );
 
+  it('should be triggered by search service to make a search', () => {
+    spyOn(component, 'onSearchQueryChangeExec');
+    searchService.triggerSearch();
+    expect(component.onSearchQueryChangeExec).toHaveBeenCalled();
+  });
+
   it('should update selection details if there are no selections', () => {
     spyOn(translateService, 'instant').and.returnValue('key');
     component.updateSelectionDetails('categories');
     expect(component.selectionDetails.categories.numSelections).toEqual(0);
   });
 
+  it('should toggle selection', () => {
+    spyOn(component, 'updateSelectionDetails');
+    spyOn(component, 'refreshSearchBarLabels');
+    component.toggleSelection('categories', 'Algorithms');
+    expect(component.updateSelectionDetails).toHaveBeenCalled();
```

Comment:
Can we use the toHaveBeenCalledWith() method instead?

Reply:
done